modules/core/asset/src/AssetStorage.php

```diff
@@ -80,14 +81,17 @@ public static function createInstance(ContainerInterface $container, EntityTypeI
    * {@inheritdoc}
    */
   protected function doPreSave(EntityInterface $entity) {
-    /** @var \Drupal\Core\Entity\ContentEntityInterface $entity */
+    /** @var \Drupal\asset\Entity\AssetInterface $entity */
     $id = parent::doPreSave($entity);
 
     // If there is no original entity, bail.
     if (empty($entity->original)) {
       return $id;
     }
 
+    // Assert field type for PHPStan checks.
+    assert($entity->get('status')->first() instanceof StateItemInterface);
+
```

Comment:
In some of our test classes I found this to be the lowest impact/most efficient way to provide additional type checking to PHPStan for custom field types that have additional properties/methods we use

Reply:
Very interesting! So simply including an `assert()` is enough to tell PHPStan what we're expecting. Makes sense!